lib/vanagon/component/dsl.rb

```diff
@@ -289,18 +289,64 @@ def environment(env)
         @component.environment.merge!(env)
       end
 
-      # Adds actions to run at the beginning of a package install
+      # Adds action to run during the preinstall phase of packaging
       #
-      # @param action [String] Bourne shell compatible scriptlets to execute
-      def add_preinstall_action(action)
-        @component.preinstall_actions << action
+      # @param pkg_state [Array] the state in which the scripts should execute. Can be
+      #   one or multiple of 'install' and 'upgrade'.
+      # @param scripts [Array] the Bourne shell compatible scriptlet(s) to execute
+      def add_preinstall_action(pkg_state, scripts)
```

Comment:
This change breaks the method interface for any users of the method previously ( I think this was just @shrug 's change). This means that we would need to tightly coordinate the vanagon release with a code update to all 3 puppet agent branches to avoid failure. This seems undesirable.